scripts/4.Covariates.py

```diff
@@ -0,0 +1,222 @@
+
+# coding: utf-8
+
+# # Incorporating covariate information into classifiers
+
+# Daniel has shown that the sample data possesses covariates in <a href='https://github.com/cognoma/cancer-data/blob/master/3.explore-mutations.ipynb'>this notebook</a>. These may provide a spurious signal that a classifier accommodates for, and could confound attempts to pick out the actual signal we desire. This notebook will create a file with encoded information on these covariates that can be used as additional data to train on.
+
+# In[1]:
+
+import pandas as pd
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+import warnings
+from sklearn import preprocessing, grid_search
+from sklearn.linear_model import SGDClassifier
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+get_ipython().magic('matplotlib inline')
+
+
+# Let's peak at the sample data.
+
+# In[2]:
+
+samples_df = pd.read_table('download/samples.tsv', index_col=0)
+samples_df.head(10)
+
+
+# Let's get a sense of how much missing data there is.
+
+# In[3]:
+
+print('Total number of samples: %d' % len(samples_df))
+print('Number of nulls in each column:')
+samples_df.isnull().sum(axis=0)
+
+
+# Now count the number of types of each column. We'll use Python's set, and therefore must exclude null values.
+
+# In[4]:
+
+categorical_variables = samples_df.columns[[1, 2, 3, 4, 6, 8]]
+for variable in categorical_variables:
+    not_null = ~samples_df[variable].isnull()
+    count = len(set(samples_df.loc[not_null, variable]))
+    print('Number of types in category %s: %d' % (variable, count))
+
+
+# There are no missing values from <code>sample_types</code> and only one possible value, so it is redundant and we'll exclude it.
+
+# In[5]:
+
+categorical_variables = categorical_variables[1:]
+print(categorical_variables)
+
+
+# Inspecting the head of the samples DataFrame above, we see that two columns, namely <code>dead</code> and <code>recurred</code>, need some attention. We're going to encode categorical variables using panda's get_dummies. Since these columns are indicated by a 1 or 0, this will become the column header when encoded, as below.
+
+# In[6]:
+
+pd.get_dummies(samples_df.dead).head(10)
+
+
+# Note that null values are encoded as (0,0) pairs... I was wondering if we were okay with this.
+# 
+# Let's rename the values in each of these so that they more accurately reflect the underlying variable.
```

Comment:
How about rename after converted to dummies, using `DataFrame.rename`?

This way you can rename for all variables at once using one big dictionary.